src/components/Popover/PopoverContent.tsx

```diff
@@ -0,0 +1,155 @@
+import styled, { css } from 'styled-components';
+import { usePopper } from 'react-popper';
+import type { Placement } from '@popperjs/core/lib/enums';
+
+import { theme } from '../../essentials/theme';
+import { get } from '../../utils/themeGet';
+import { Spaces } from '../../essentials';
+
+import { Card } from '../Card/Card';
+
+import { transition, useClickOutside } from './utils';
+
+interface PopoverContentProps {
+    /**
+     * Popover content (can be any valid React Element or component)
+     */
+    children: React.ReactNode;
+    /**
+     * Popover Trigger Reference
+     */
+    triggerRef: HTMLElement | HTMLDivElement | null;
+    /**
+     * Is Popover currently displayed?
+     */
+    isShown: boolean;
+    /**
+     * Optional: Specify the Popover content placement (placement changes automatically, if Popover content cannot fit inside viewport with selected placement)
+     */
+    placement?: Placement;
+    /**
+     * Optional: Specify the Popover content offset (margin between Popover trigger and content)
+     */
+    offset?: number;
+    /**
+     * Passes the onClose callback
+     */
+    onClose: (event?: React.KeyboardEvent<HTMLDivElement>) => void;
+}
+
+interface StyledPopoverProps extends React.CSSProperties {
+    isShown: boolean;
+    transform?: string;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    translate?: any;
+    top?: string | number | undefined;
+    left?: string | number | undefined;
+    bottom?: string | number | undefined;
+    right?: string | number | undefined;
+    onKeyDown: (event: React.KeyboardEvent<HTMLDivElement>) => void;
+    ref: MutableRefObject<HTMLDivElement>;
+    tabIndex: number;
+    role: string;
+}
+
+const StyledPopoverParent = styled.div.attrs({ theme })<StyledPopoverProps>`
+    ${({ isShown, transform, top, left, bottom, right, position }) => css`
+        box-sizing: border-box;
+        position: ${position};
+        top: ${top};
+        left: ${left};
+        bottom: ${bottom};
+        right: ${right};
+        transform: ${transform};
+        transition: ${transition(['opacity'], '0.15s', 'ease-in-out')};
+        width: auto;
+        height: auto;
+        z-index: 1000;
+        box-shadow: ${get('shadows.small')};
+        max-height: none;
+        opacity: ${isShown ? '1' : '0'};
+
+        &:focus {
+            outline: 0;
+        }
+    `}
+`;
+
+const PopoverContentContainer = styled(Card)`
+    display: block;
+    padding: ${Spaces[2]};
+`;
+
+export const PopoverContent = ({
+    children,
+    isShown,
+    onClose,
+    offset,
+    placement,
+    triggerRef
+}: PopoverContentProps): JSX.Element => {
+    const popoverRef = React.useRef<HTMLDivElement | null>(null);
+
+    const { styles, update } = usePopper(triggerRef, popoverRef.current, {
+        placement,
+        strategy: 'fixed',
+        modifiers: [
+            {
+                name: 'offset',
+                enabled: !!offset,
```

Comment:
```suggestion
                enabled: Boolean(offset)
```
Wdyt?

Reply:
actually I think this just needed to be **true** by default, as we have a default offset, but good suggestion!